docs/topics/functions.md

```diff
@@ -142,53 +198,43 @@ skipped argument, you must name all subsequent arguments:
 reformat("This is a short String!", upperCaseFirstLetter = false, wordSeparator = '_')
 ```
 
-You can pass a [variable number of arguments (`vararg`)](#variable-number-of-arguments-varargs) with names using the
-_spread_ operator (prefix the array with `*`):
+You can pass a [variable number of arguments](#variable-number-of-arguments-varargs) (`vararg`) naming the correspoding array:
 
 ```kotlin
 fun foo(vararg strings: String) { /*...*/ }
 
-foo(strings = *arrayOf("a", "b", "c"))
+foo(strings = arrayOf("a", "b", "c"))
 ```
 
+<!-- Rationale for named arguments interaction with varargs is here https://youtrack.jetbrains.com/issue/KT-52505#focus=Comments-27-6147916.0-0 -->
+
 > When calling Java functions on the JVM, you can't use the named argument syntax because Java bytecode does not
 > always preserve the names of function parameters.
 >
 {style="note"}
 
 ### Unit-returning functions
 
-If a function does not return a useful value, its return type is `Unit`. `Unit` is a type with only one value - `Unit`.
-This value does not have to be returned explicitly:
+If a function does not return a useful value, its return type is `Unit` (corresponds to the `void` type in Java).
+
+`Unit` is a type with only one value - `Unit`.
+You don't have to declare this return type, or return `Unit` explicitly.
+Therefore, this verbose declaration:
 
 ```kotlin
 fun printHello(name: String?): Unit {
     if (name != null)
         println("Hello $name")
     else
         println("Hi there!")
-    // `return Unit` or `return` is optional
+    return Unit
 }
 ```
 
-The `Unit` return type declaration is also optional. The above code is equivalent to:
-
-```kotlin
-fun printHello(name: String?) { ... }
-```
-
-### Single-expression functions
-
-When the function body consists of a single expression, the curly braces can be omitted and the body specified after an `=` symbol:
-
-```kotlin
-fun double(x: Int): Int = x * 2
-```
-
-Explicitly declaring the return type is [optional](#explicit-return-types) when this can be inferred by the compiler:
+is equivalent to:
 
 ```kotlin
-fun double(x: Int) = x * 2
+fun printHello(name: String?) { /*...*/ }
```

Comment:
Please include the code in `...`, because otherwise it's difficult to understand what stays and what's gone.

```suggestion
fun printHello(name: String?) {
    if (name != null)
        println("Hello $name")
    else
        println("Hi there!")
}
```